lib/ecto/exceptions.ex

```diff
@@ -117,3 +117,18 @@ end
 defmodule Ecto.MigrationError do
   defexception [:message]
 end
+
+defmodule Ecto.StaleModelError do
+  defexception [:message]
+
+  def exception(opts) do
+    action = Keyword.fetch!(opts, :action)
+    model = Keyword.fetch!(opts, :model)
+
+    msg = """
+    attempted to #{action} a stale #{model}
```

Comment:
About that, what do you think we should print in the exception message? This just prints the model's module, which may not be ideal.